src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,242 @@
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, payload.RootElement.GetString("user_id"), ClaimValueTypes.String, Options.ClaimsIssuer));
+
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        protected override string FormatScope() => string.Join(",", Options.Scope);
+
+        /// <summary>
+        /// Gets an HTTP payload as an asynchronous operation.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        /// <param name="elementName">Name of the element.</param>
+        private static async Task<string> GetRawStringAsync(
+            [NotNull] HttpResponseMessage message,
+            [NotNull] string elementName)
+        {
+            var messageStream = await message.Content.ReadAsStreamAsync();
+
+            var document = await JsonSerializer.DeserializeAsync<JsonElement>(messageStream);
+            string elementContent = document.GetString(elementName);
+
+            return elementContent;
+        }
+
+        /// <summary>
+        /// Gets the RSA2 signature.
+        /// </summary>
+        /// <param name="source">The source.</param>
+        /// <param name="clientSecret">The client secret.</param>
+        private static string GetRSA2Signature(
+            [NotNull] SortedDictionary<string, string> source,
+            [NotNull] string clientSecret)
+        {
+            var builder = new StringBuilder();
+
+            foreach (var pair in source)
+            {
+                if (string.IsNullOrEmpty(pair.Value))
+                {
+                    continue;
+                }
+
+                builder.Append(pair.Key)
+                       .Append('=')
+                       .Append(pair.Value)
+                       .Append('&');
+            }
+
+            builder.Remove(builder.Length - 1, 1);
+
+            byte[] dataBytes = Encoding.UTF8.GetBytes(builder.ToString());
+
+            using var privateKeyRsaProvider = CreateAlgorithm(clientSecret);
+            byte[] signatureBytes = privateKeyRsaProvider.SignData(dataBytes, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+
+            return Convert.ToBase64String(signatureBytes);
+        }
+
+        /// <summary>
+        /// Creates the algorithm.
+        /// </summary>
+        /// <param name="clientSecret">The client secret.</param>
+        private static RSA CreateAlgorithm([NotNull] string clientSecret)
```

Comment:
You could make this non-static and just pass in the value of `Options.ClientSecret` on line 227?

```csharp
byte[] privateKeyBytes = Convert.FromBase64String(Options.ClientSecret);
```